packages/generator/templates/helpers.ts.hbs

```diff
@@ -2,31 +2,67 @@
 import { ShopperToken, stripBearer } from "@commerce-apps/core"
 import  * as sdk from "./"
 
-
-const shopperAuthApi = sdk.{{shopperAuthClient}}.Client.prototype.{{shopperAuthApi}};
+//type of the input parameter of auth function
+type authFuncParamType = Parameters<typeof sdk.{{shopperAuthClient}}.Client.prototype.{{shopperAuthApi}}>[0];
 
 /**
  * @description This wraps the parameters for the authCall to retrieve a token. 
- *  That call can be called directly, this is here just for convenience
- * @export
- * @template T A template used to extend the parameters of the shopper authentication call
+ * That call can be made directly, this is here just for convenience
+ *
  * @param clientConfig Client configuration properties
- * @param options Properties to get the auth token. Properties that are already set in clientConfig can be omitted
+ * @param body Post body required for authorization
  * @returns {Promise<ShopperToken>}
  */
+export async function getShopperAuthToken (
+    clientConfig: sdk.ClientConfig,
+    body: authFuncParamType["body"]
+): Promise<ShopperToken>{
+    let client = new sdk.{{shopperAuthClient}}.Client(clientConfig);
+
+    let response = (await client.{{shopperAuthApi}}({
+        rawResponse: true,
+        body: body
+    })) as Response;
+    if (!response.ok) {
+        throw new Error(`${response.status}: ${response.statusText}`);
+    }
+    return new ShopperToken(stripBearer(response.headers.get("Authorization")));
+}
+
+const shopperAuthApi = sdk.{{shopperAuthClient}}.Client.prototype.{{shopperAuthApi}};
+
+/**
+* @deprecated Use the "getShopperAuthToken" function instead
+* @description This wraps the parameters for the authCall to retrieve a token.
+*  That call can be called directly, this is here just for convenience
+* @export
+* @template T A template used to extend the parameters of the shopper authentication call
+* @param {(T extends (arg: infer A) => any
+*           ? { parameters: { clientId?: string; shortCode?: string } } & A
+*           : never)} options This provides the type ahead for this parameter properly
+* @returns {Promise<ShopperToken>}
+*/
 export async function getAuthToken<T extends typeof shopperAuthApi>(
-        clientConfig: sdk.ClientConfig,
-        options: T extends (arg: infer A) => any ? A : never
-    ): Promise<ShopperToken>{
+    options: T extends (arg: infer A) => any
+    ? { parameters: { clientId?: string; shortCode?: string } } & A
+    : never
+    ): Promise<ShopperToken> {
```

Comment:
Can we just make this a call to function above?

Reply:
Since its deprecated I prefer to leave as it is